rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +83,293 @@ typedef struct
+        _info._img_size = std::move(img_size);
+        _mask_cords = std::move(mask_cords);
+        _polygon_count = std::move(polygon_count);
+        _vertices_count = std::move(vertices_count);
     }
-    void set_bb_cords_xcycwh(BoundingBoxCords_xcycwh bb_cords_xcycwh) { _bb_cords_xcycwh =std::move(bb_cords_xcycwh); }
-    void set_bb_labels(BoundingBoxLabels bb_label_ids) { _bb_label_ids = std::move(bb_label_ids); }
-    void set_img_size(ImgSize img_size) { _img_size = std::move(img_size); }
+    std::vector<int>& get_polygon_count() { return _polygon_count; }
+    std::vector<std::vector<int>>& get_vertices_count() { return _vertices_count; }
+    MaskCords& get_mask_cords() { return _mask_cords;}
+    void set_mask_cords(MaskCords mask_cords) { _mask_cords = std::move(mask_cords); }
+    void set_polygon_counts(std::vector<int> polygon_count) { _polygon_count = std::move(polygon_count); }
+    void set_vertices_counts(std::vector<std::vector<int>> vertices_count) { _vertices_count = std::move(vertices_count); }
+    protected:
+    MaskCords _mask_cords = {};
+    std::vector<int> _polygon_count = {};
+    std::vector<std::vector<int>> _vertices_count = {};
 };
 
-struct KeyPoint : public MetaData
+struct KeyPoint : public BoundingBox
 {
     KeyPoint()= default;
     KeyPoint(ImgSize img_size, JointsData *joints_data)
     {
-        _img_size = std::move(img_size);
+        _info._img_size = std::move(img_size);
         _joints_data = std::move(*joints_data);
     }
     void set_joints_data(JointsData *joints_data) { _joints_data = std::move(*joints_data); }
+    JointsData& get_joints_data(){ return _joints_data; }
+    protected:
+    JointsData _joints_data = {};
+};
+
+struct MetaDataDimensionsBatch
+{
+    std::vector<std::vector<size_t>>& labels_dims() { return _labels_dims; }
+    std::vector<std::vector<size_t>>& bb_cords_dims() { return _bb_coords_dims; }
+    std::vector<std::vector<size_t>>& mask_cords_dims() { return _mask_coords_dims; }
+    void clear()
+    {
+        _labels_dims.clear();
+        _bb_coords_dims.clear();
+        _mask_coords_dims.clear();
+    }
+    void resize(size_t size)
+    {
+        _labels_dims.resize(size);
+        _bb_coords_dims.resize(size);
+        _mask_coords_dims.resize(size);
+    }
+    void insert(MetaDataDimensionsBatch &other)
+    {
+        _labels_dims.insert(_labels_dims.end(), other.labels_dims().begin(), other.labels_dims().end());
+        _bb_coords_dims.insert(_bb_coords_dims.end(), other.bb_cords_dims().begin(), other.bb_cords_dims().end());
+        _mask_coords_dims.insert(_mask_coords_dims.end(), other.mask_cords_dims().begin(), other.mask_cords_dims().end());
+    }
+private:
+    std::vector<std::vector<size_t>> _labels_dims = {};
+    std::vector<std::vector<size_t>> _bb_coords_dims = {};
+    std::vector<std::vector<size_t>> _mask_coords_dims = {};
+};
+
+struct MetaDataInfoBatch {
+public:
+    std::vector<uint> _img_ids = {};
```

Comment:
Public variables can be initialized without _ operator

Reply:
Done